packages/app-builder-lib/src/node-module-collector/nodeModulesCollector.ts

```diff
@@ -1,37 +1,79 @@
+    const tree = await this.getDependenciesTree()
+    const realTree = this.getTreeFromWorkspaces(tree)
+    const parsedTree = this.extractDependencyTree(realTree)
+
+    this.collectAllDependencies(parsedTree)
+
+    const productionTree = this.removeNonProductionDependencies(parsedTree)
+    const dependencyGraph = this.convertToDependencyGraph(productionTree)
+
+    const hoisterResult = hoist(this.transToHoisterTree(dependencyGraph), { check: true })
+    this._getNodeModules(hoisterResult.dependencies, this.nodeModules)
+
+    return this.nodeModules
   }
 
-  private transToHoisterTree(obj: DependencyGraph, key: string = `.`, nodes: Map<string, HoisterTree> = new Map()): HoisterTree {
-    let node = nodes.get(key)
-    const name = key.match(/@?[^@]+/)![0]
-    if (!node) {
-      node = {
-        name,
-        identName: name,
-        reference: key.match(/@?[^@]+@?(.+)?/)![1] || ``,
-        dependencies: new Set<HoisterTree>(),
-        peerNames: new Set<string>([]),
-      }
-      nodes.set(key, node)
+  protected abstract getCommand(): string
+  protected abstract getArgs(): string[]
+  protected abstract parseDependenciesTree(jsonBlob: string): NpmDependency
+  protected abstract removeNonProductionDependencies(parsedTree: DependencyTree): DependencyTree
 
-      for (const dep of (obj[key] || {}).dependencies || []) {
-        node.dependencies.add(this.transToHoisterTree(obj, dep, nodes))
-      }
+  protected async getDependenciesTree(): Promise<NpmDependency> {
+    const command = this.getCommand()
+    const args = this.getArgs()
+    const dependencies = await exec(command, args, {
+      cwd: this.rootDir,
+      shell: true,
+    })
+    return this.parseDependenciesTree(dependencies)
+  }
+
+  private extractDependencyTree(npmTree: NpmDependency): DependencyTree {
+    const { name, version, path, workspaces, dependencies, _dependencies, optionalDependencies, peerDependencies } = npmTree
+    const tree: DependencyTree = {
+      name,
+      version,
+      path,
+      workspaces,
+      implicitDependenciesInjected: false,
     }
-    return node
+
+    const extractInternal = (deps: NpmDependency["dependencies"]) =>
+      deps && Object.keys(deps).length > 0
+        ? Object.entries(deps).reduce((accum, [packageName, depObjectOrVersionString]) => {
+            return {
+              ...accum,
+              [packageName]:
+                typeof depObjectOrVersionString === "object" && Object.keys(depObjectOrVersionString).length > 0
+                  ? this.extractDependencyTree(depObjectOrVersionString as any)
+                  : depObjectOrVersionString,
+            }
+          }, {})
+        : undefined
+
+    // only set property if existing, skip undefined (at minimum, it helps in Variables debugger window)
+    use(_dependencies, v => (tree._dependencies = v))
+    use(optionalDependencies, v => (tree.optionalDependencies = v))
+    use(peerDependencies, v => (tree.peerDependencies = v))
```

Comment:
We can delete `peerDependencies`. There is no place referencing this peerDependencies, so we can remove it.